File: MultiLanguageSandbox/src/test/deps/java/13.java
import java.util.HashMap;
import java.util.Map;
import java.util.Map;
import java.util.HashMap;
import java.util.Arrays;

class Solution {
    /**
     * Given an array of integers nums and an integer target,
     * find two numbers such that they add up to target and return their indices.
     * You may assume that each input would have exactly one solution,
     * and you may not use the same element twice.
     * You can return the answer in any order.
     *
     * Example 1:
     * Input: nums = [2,7,11,15], target = 9
     * Output: [0,1]
     * Explanation: Because nums[0] + nums[1] == 9, we return [0, 1].
     *
     * Example 2:
     * Input: nums = [3,2,4], target = 6
     * Output: [1,2]
     *
     * Example 3:
     * Input: nums = [3,3], target = 6
     * Output: [0,1]
     */
  public int[] twoSum(int[] nums, int target) 
{
        Map<Integer, Integer> numMap = new HashMap<>();
        
        for (int i = 0; i < nums.length; i++) {
            int complement = target - nums[i];
            if (numMap.containsKey(complement)) {
                return new int[] { numMap.get(complement), i };
            }
            numMap.put(nums[i], i);
        }
        
        // The problem states that there is exactly one solution, so this line is theoretically unreachable
        throw new IllegalArgumentException("No two sum solution");
    }
  public static void main(String[] args) {
    Solution solution = new Solution();

    // Test the function with different test cases
    assert Arrays.equals(
      solution.twoSum(new int[] { 1, 3, 7, 9, 2 }, 11),
      new int[] { 3, 4 }
    );
    assert Arrays.equals(
      solution.twoSum(new int[] { 1, 5, 3, 7, 8 }, 9),
      new int[] { 0, 4 }
    );
    assert Arrays.equals(
      solution.twoSum(new int[] { 10, 20, 30, 40, 50 }, 90),
      new int[] { 3, 4 }
    );
    System.out.println("All tests passed");
  }
}

File: MultiLanguageSandbox/src/test/deps/ruby/13.rb


# Finds the maximum sum of any three numbers from a given array that is divisible by a specified number k.
# This function takes an array of numbers and an integer k as arguments. It returns the maximum sum 
# of any three numbers from the array such that this sum is divisible by k. It's guaranteed that there 
# are at least three numbers in the array and that a solution exists.
#
# Example:
# >>> max_sum_divisible_by_k([1, 2, 3, 4], 3)
# 9
# >>> max_sum_divisible_by_k([4, 5, 10, 20], 5)
# 35

def max_sum_divisible_by_k(numbers, k)
  max_sum = -Float::INFINITY
  # Generate all possible combinations of three numbers
  numbers.combination(3).each do |a, b, c|
    current_sum = a + b + c
    if current_sum % k == 0 && current_sum > max_sum
      max_sum = current_sum
    end
  end
  max_sum
end

raise 'Test failed' unless max_sum_divisible_by_k([1, 2, 3, 4], 3) == 9
raise 'Test failed' unless max_sum_divisible_by_k([4, 5, 10, 20], 5) == 35
raise 'Test failed' unless max_sum_divisible_by_k([6, 12, 17, 23, 29], 7) == 35
raise 'Test failed' unless max_sum_divisible_by_k([7, 14, 21, 28], 7) == 63
raise 'Test failed' unless max_sum_divisible_by_k([10, 15, 25, 30, 40], 10) == 80


puts 'All tests passed!'